From bd1a9cfa437306562f105e5d7d6519cbff162e24 Mon Sep 17 00:00:00 2001
From: time-river <me@vvl.me>
Date: Mon, 5 Feb 2018 00:18:43 +0800
Subject: [PATCH 3/5] modify som unsutable code

Signed-off-by: time-river <me@vvl.me>
---
 src/locking/lock_driver_dlm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/locking/lock_driver_dlm.c b/src/locking/lock_driver_dlm.c
index 8bfad858c..66111af8a 100644
--- a/src/locking/lock_driver_dlm.c
+++ b/src/locking/lock_driver_dlm.c
@@ -49,7 +49,7 @@
 #define DLM_LOCKSPACE_NAME  "libvirt"
 
 #define LOCK_RECORD_FILE_MODE       0644
-#define LOCK_RECORD_FILE_PATH       "/tmp/dlmfile"
+#define LOCK_RECORD_FILE_PATH       "/tmp/libvirtd-dlm-file"
 
 #define PRMODE  "PRMODE"
 #define EXMODE  "EXMODE"
@@ -147,6 +147,7 @@ static int virLockManagerDlmLoadConfig(const char *configFile)
     if (virConfGetValueBool(conf, "auto_disk_leases", &driver->autoDiskLease) < 0)
         goto cleanup;
 
+    driver->requireLeaseForDisks = !driver->autoDiskLease;
     if (virConfGetValueBool(conf, "require_lease_for_disks", &driver->requireLeaseForDisks) < 0)
         goto cleanup;
 
-- 
2.11.0

