From 4b918f956cf65838ea9b6b000bff4e8a883cd607 Mon Sep 17 00:00:00 2001
From: time-river <me@vvl.me>
Date: Mon, 5 Feb 2018 00:44:22 +0800
Subject: [PATCH 5/5] delete useless content

Signed-off-by: time-river <me@vvl.me>
---
 src/Makefile.am | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index c95507532..95b46dc89 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3307,9 +3307,6 @@ endif WITH_LIBVIRTD
 	$(MKDIR_P) "$(DESTDIR)$(localstatedir)/lib/libvirt/images"
 	$(MKDIR_P) "$(DESTDIR)$(localstatedir)/lib/libvirt/filesystems"
 	$(MKDIR_P) "$(DESTDIR)$(localstatedir)/lib/libvirt/boot"
-if WITH_DLM
-       $(MKDIR_P) "$(DESTDIR)$(localstatedir)/lib/libvirt/dlm"
-endif WITH_DLM
 if WITH_SANLOCK
 	$(MKDIR_P) "$(DESTDIR)$(localstatedir)/lib/libvirt/sanlock"
 endif WITH_SANLOCK
-- 
2.11.0

